app/jobs/import_versions_job.rb

```diff
@@ -102,6 +111,13 @@ def import_record(record)
     version.update_different_attribute(save: false)
     version.save
 
+    if version.uuid_previously_changed?
```

Comment:
You should just be able to check `existing` here, which would be a bit more clear. Or am I missing something?

```suggestion
    unless existing
```

Reply:
✅ 

I also renamed `existing` to be `existing_version`